### ISSUE TITLE
Training parameters for Bonito: benchmarking

### ISSUE THREAD

**Issue by TimD1** [nanoporetech/bonito#18]
Hi @iiSeymour 
First of all, thanks for all the great work you've done on Bonito: keep it up!

I've been exploring different methods for improving basecalling accuracy through pre-processing, and am reaching the point where I would like to compare directly against current state-of-the-art models. According to the "Nanopore Community" website, Bonito achieves 96.16% modal raw read accuracy, and I've seen similar results when using it myself. Would you be willing to share details regarding how the final model was trained and evaluated so that I can replicate it, and accurately compare my own techniques?

For example, I would like to know:
- **training set** (I believe you used the standard HDF5 file included in the repo, but how many **chunks**, and what **min/max sequence length**?)
- **batch size**
- **test set**
Did you find using **half-precision** or different **learning rates** to have much effect on accuracy? How many **epochs** did it take for training to converge, and how long was that on your GPU cluster?

I could do some hyperparameter searching myself, but if you'd be willing to share just the parameters chosen for the final model it would save me a lot of time and I'd greatly appreciate it. Thanks in advance for your help!

**Comment by iiSeymour** [comment]
Hey @TimD1 

Thanks, I'd be interested to hear more about your preprocessing ideas.

 - **Training** - I usually set 1,000,000 chunks. I can't remember exactly the chunking parameters used as I have played around with a few different strategies. I feel an optimal chunking strategy is still a bit of an open question. The other training parameters are stored in the `config.toml` so you will find the learning rate and batch size in there *(the released model was trained on a DGX-2 so the batch size is very high)*. Mixed-precision training and half-precision inference seem to make no difference and therefore is always worth it _i.m.o_. Bonito is using `AdadW` with a `CosineAnnealingLR` scheduler for the learning rate so it's robust in that regard. Training is much slower than I would like _a.t.m_ however one thing I suggest you try, is using the released model as a starting point for your experiments.

 - **Testing** We have a few mixed sets that we use for evaluation *(and we are aiming to make these publically available)*. The evaluation pipeline I use is very simple, align the basecalls with `minimap2` then using `stats_from_bam` from [pomoxis](https://github.com/nanoporetech/pomoxis) which will give you a csv file containing the accuracy per read among other metrics. 

See `bonito tune` if you do want to do any form of hyperparameter searching. If you have access to multi-gpu systems it can do a distributed search.

**Comment by TimD1** [comment]
Thanks for the clarification and advice! If things start working well, I'll be sure to keep you in the loop.
